[BUG] Parsing Exception on FLINK SQL when registering catalogs

**Describe the [bug**](url)
When trying to use a tenant/namespace as a database seems like im getting a parsing exception

**To Reproduce**
```
Flink SQL> SHOW CATALOGS;
default_catalog
pulsarcatalog

Flink SQL> USE CATALOG pulsarcatalog;

Flink SQL> SHOW DATABASES;
public/default
public/functions
pulsar/system
sample/ns1

Flink SQL> USE `public/default`;
[ERROR] Could not execute SQL statement. Reason:
org.apache.flink.sql.parser.impl.ParseException: Encountered "/" at line 1, column 11.
Was expecting one of:
    <EOF>
    "." ...


Flink SQL> USE 'public/default';
[ERROR] Could not execute SQL statement. Reason:
org.apache.flink.sql.parser.impl.ParseException: Incorrect syntax near the keyword 'USE' at line 1, column 1.
Was expecting one of:
    "ABS" ...
    "ALTER" ...
    "ARRAY" ...
```

--- a/pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/PulsarMetadataReader.java
+++ b/pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/PulsarMetadataReader.java
@@ -190,8 +190,19 @@ public class PulsarMetadataReader implements AutoCloseable {
     public void deleteTopic(String topicName) throws PulsarAdminException {
         int partitionNum = admin.topics().getPartitionedTopicMetadata(topicName).partitions;
         if (partitionNum > 0) {
+            final Optional<PersistentTopicInternalStats> any = admin.topics().getPartitionedInternalStats(topicName).partitions.entrySet()
+                .stream()
+                .map(Map.Entry::getValue)
+                .filter(p -> !p.cursors.isEmpty())
+                .findAny();
+            if (any.isPresent()) {
+                throw new IllegalStateException(String.format("The topic[%s] cannot be deleted because there are subscribers", topicName));
+            }
             admin.topics().deletePartitionedTopic(topicName, true);
         } else {
+            if (!admin.topics().getInternalStats(topicName).cursors.isEmpty()) {
+                throw new IllegalStateException(String.format("The topic[%s] cannot be deleted because there are subscribers", topicName));
+            }
             admin.topics().delete(topicName, true);
         }
     }

--- a/pulsar-flink-connector/src/main/java/org/apache/flink/table/catalog/pulsar/PulsarCatalog.java
+++ b/pulsar-flink-connector/src/main/java/org/apache/flink/table/catalog/pulsar/PulsarCatalog.java
@@ -267,7 +267,7 @@ public class PulsarCatalog extends GenericInMemoryCatalog {
             if (!ignoreIfNotExists) {
                 throw new TableNotExistException(getName(), tablePath, e);
             }
-        } catch (PulsarAdminException e) {
+        } catch (PulsarAdminException | RuntimeException e) {
             throw new CatalogException(String.format("Failed to drop table %s", tablePath.getFullName()), e);
         }
     }

